crates/net/network/src/manager.rs

```diff
@@ -675,7 +675,10 @@ where
         // If the budget is exhausted we manually yield back control to the (coop) scheduler. This
         // manual yield point should prevent situations where polling appears to be frozen. See also <https://tokio.rs/blog/2020-04-preemption>
         // And tokio's docs on cooperative scheduling <https://docs.rs/tokio/latest/tokio/task/#cooperative-scheduling>
-        let mut budget = 1024;
+        //
+        // Testing has shown that this loop naturally reaches the pending state within 1-5
```

Comment:
does testing include fuzzing here? otherwise please open an issue to do this additionally

Reply:
even though the budget is 5, polling `Swarm` may block:
while loops can block
https://github.com/paradigmxyz/reth/blob/29701f6486717514673e61b2b9b49418de7c4187/crates/net/network/src/swarm.rs#L312-L316

just continues the outer loop endlessly if there is io ready
https://github.com/paradigmxyz/reth/blob/29701f6486717514673e61b2b9b49418de7c4187/crates/net/network/src/swarm.rs#L325
https://github.com/paradigmxyz/reth/blob/29701f6486717514673e61b2b9b49418de7c4187/crates/net/network/src/swarm.rs#L336

hence why we need to fuzz this or unblock swarm

Reply:
goo idea, we already have the Testnet struct, so we could implement this easily